apps/tandoor/docker-compose.yml

```diff
@@ -0,0 +1,68 @@
+version: "3.7"
+services:
+  tandoor:
+    container_name: tandoor
+    image: ghcr.io/tandoorrecipes/recipes:1.5.4
+    volumes:
+      - ${APP_DATA_DIR}/data/staticfiles:/opt/recipes/staticfiles
+      - ${APP_DATA_DIR}/data/mediafiles:/opt/recipes/mediafiles
+    ports:
+      - ${APP_PORT}:8080
+    environment:
+      - SECRET_KEY=${TANDOOR_SECRET_KEY}
+      - DB_ENGINE=django.db.backends.postgresql
+      - POSTGRES_HOST=tandoor-db
+      - POSTGRES_PORT=5432
+      - POSTGRES_USER=tandoor
+      - POSTGRES_PASSWORD=${TANDOOR_POSTGRESS_PASSWORD}
+      - POSTGRES_DB=tandoordb
+    networks:
+      - tipi_main_network
+    restart: unless-stopped
+    labels:
+      # Main
+      traefik.enable: true
+      traefik.http.middlewares.tandoor-web-redirect.redirectscheme.scheme: https
+      traefik.http.services.tandoor.loadbalancer.server.port: 8341
```

Comment:
I can't set the app port to 8080. There are other apps running on there.

Reply:
that's not a problem, here is the internal port used in the container to use by traefik to serve on https.
So you can have many dockers with the same instruction.
I tried with my tipi install without issues.
The only part that is referencing the external port is "ports" on main app only.
Traefik is a reverse proxy, which expose on 443/80 only. But it needs to know the internal port to redirect traffic.

Reply:
Yeah but if I am not using SSL and connecting directly?

Reply:
I say that because in for example dashdot is like I have configured it here and it works with ssl too when I tried it. So it should be fine.

Reply:
Not a problem, it's only used for people using "tipi custom domain".
You can take a look here: https://github.com/meienberger/runtipi-appstore/pull/1000/files#diff-8084417fe3bb5a8e0c793fb1d8ce89951452cfe2b9fa5595fd92b5b133a7e89fR30
If you are not using SSL, traefik is not used for you as you are accessing your app directly with the port.

Reply:
Yeah but if I set it to 8080 it will listen on 8080 without SSL. I don't want it to do that. All the apps have a random port like this so this is how we should keep it (I thing).

Reply:
How it's working to access your app:

1. You -> (SECURE, but self-signed certificate) https://tandoor.tipi.lan (:443) -> traefik -> tandoor:8080
2. You -> http://tandoor.tipi.lan:8341 -> tandoor:8341
3. You -> (SECURE) https://yourdomain.com (:443) -> traefik -> tandoor:8080

I suppose that you are using scenario 2, wich is not ssl and only http.

Scenario 1 & 3 will only work with:
`traefik.http.services.tandoor.loadbalancer.server.port: 8080`
If port is `8341`, traefik will try to reach app in tandoor container at port `8341` which is wrong, because tandoor container is only exposing app internally on `8080`.

Scenario 2 is working because:
```
ports:
      - ${APP_PORT}:8080
```
You are exposing directly the app without traefik and no ssl on port 8080.
Also I doubt that port is not already used by another app in app db.
I you want to set a custom port, it's on this line.
And traefik needs always to know the binding port of internal app (which is currently 8080)

Reply:
You can take as example the recent PR from @DireMunchkin here: https://github.com/meienberger/runtipi-appstore/pull/998

Reply:
But how [this](https://github.com/meienberger/runtipi-appstore/blob/master/apps/sshwifty/docker-compose.yml) works?

Reply:
> But how [this](https://github.com/meienberger/runtipi-appstore/blob/master/apps/sshwifty/docker-compose.yml) works?

Simple, it don't work, I think that @meienberger missed it during review of the PR.
Just tried it on my install and scenario 1/3 are not working, I'm facing a "Bad gateway", but it's logical as traefik try to reach app at sshwifty:8273 which is wrong as app is exposed internally on 8182.

Reply:
Is there any way to use a different port rather than 8080 tho?

Reply:
Some app like SFTPGo can be tweaked to select the binding port: https://github.com/meienberger/runtipi-appstore/pull/973/files#diff-7722e3228274b6a6c69001a28734221bd69f64d2528768c8cae4cfb20b1ad652R27
But most of the time it's not possible. Also it will be clear all the time even if you change the port, otherwise you need to setup SSL in app itself to secure communication between traefik and container, but that a real headhache to manage N*Apps certificates. That's why we recommend using traefik, because communication between you and treafik is secured, so your app is securely exposed. Only traffic between your app and traefik is clear, but as it's internal, it's not a problem.

Reply:
But the port is app-port:8080 so if traefik asks for something like this app-port:8143 where is the problem. That's how it works right?

Reply:
Not really, ports is exposing the container directly on a port of the host. Traffic on the other side is grabbing the app on an internal docker network, the "main_tipi_network" that you see in each docker compose files. The docker network permit all containers in it to communicate between them without being exposed publicly ;)

Reply:
So can't trafik use port 8080 but expose port 8143 for no SSL?

Reply:
To give an example a lot of apps expose port 80 how do they work with no SSL?

Reply:
Yes, but for that you need to resolve my comment to tell traefik to use 8080.
Port 8143 will be exposed directly by docker with the "ports" instruction. It's already effective.
Don't hesitate to read further about docker and traefik as reverse proxy, can be complicated at start, but in fact it's really easy once your understand how all things are working together ;)

Reply:
So how do I so that because I really can't use port 8080...

Reply:
Try what I suggest on original comment, you will see that it will not change anything for you

Reply:
What is the original comment? We had a loooong conversation...

Reply:
Change the line:

`traefik.http.services.tandoor.loadbalancer.server.port: 8341`

To

`traefik.http.services.tandoor.loadbalancer.server.port: 8080`

And you can try after that ;)

Reply:
But this doesn't expose the app to port 8080 which I can't expose?

Reply:
Nop, app will remain exposed on your custom port. Try it you will see

Reply:
But won't I have to uninstall it and reinstall it to apply the changes?

Reply:
depends where you are editing files, if it's under runtipi/apps a stop/start will be fine

Reply:
But when it comes to updating it will reset it. So I will set port 8080 now and also fix sshwifty...

Reply:
So I understood how it works!!! (I think). So the environment app variable exposes the port to the preconfigured port inside the config.json file but the traefik port just tells traefik to use port 8080 that the app is running on without exposing. Please tell me that's right..